### ISSUE TITLE
`--preload-size` not playing well with multiple databases

### ISSUE THREAD

**Issue by jvolkening** [fbreitwieser/krakenuniq#157]
In testing the use of `--preload-size` in our pipelines (KrakenUniq v1.0.4), I've noticed the following issue:

We regularly use multiple databases in our KrakenUniq calls, e.g. `--db db1 --db db2 --db db3`, etc. The largest of the individual databases are ~ 110 GB. I can successfully run `krakenuniq` using `--preload-size 40G` on a 128GB server using one of these databases, but when I try to run the same search using two or more databases, KrakenUniq eventually exits silently with empty output files, no error, and an exit code of `0`.

I'm not entirely sure how to debug this -- the best I've done is by watching resource usage as the following command runs:
```bash
krakenuniq --db db1 --db db2 --threads 16 --report-file k.report --output k.results --preload-size 40G --paired R1.fq R2.fq
```
What I observe when run with a *single* 110 GB database is that KrakenUniq slowly fills up RAM to about the requested 40GB with CPU usage at ~ 10%, then there is a spike in CPU usage and on STDOUT `Processed 80000 sequences (database chunk 1 of 3)`. This happens two more times for the other two chunks with memory usage steady at 40GB, and the run wraps up cleanly.

If I run with *two databases* each ~110 GB, I see the same behavior initially. Then, after it reports `Processed 80000 sequences (database chunk 3 of 3)`, memory starts to steadily creep up to ~ 80% of installed, and then KrakenUniq silently exits (code 0) with empty output files and a bunch of temporary files left around in the working directory.

If there's any more information I can provide or more formal debugging I can do, please let me know. I should mention that I can successfully search these same multiple databases (we use nine individual databases in production) if I leave out the `--preload-size` option.

**Comment by salzberg** [comment]
hi Jeremy,
You've managed to come up with a use case for the --preload-size option that we didn't anticipate! I'm afraid that KrakenUniq simply doesn't handle this case - when you use preload-size, you can't use multiple databases. I talked to the guys who wrote the code - primarily Christopher Pockrandt and also Aleksey Zimin - and they tell me there's no simple way to add this capability. The preload-size option tells KrakenUniq to break the database (just 1 db) into chunks that are no larger than N, where N is the size you specify, and read the input file multiple times while comparing it to each chunk. 
  With more than 1 db, this would become much more complex. We're going to update the docs and explain that preload-size only permits 1 db file, and we'll also update the wrapper scripts to return an error message if someone calls it with options that it can't handle.

**Comment by jvolkening** [comment]
Okay, thanks for the explanation. We'll keep doing things the way we have been -- the multiple database ability is just so very useful. I see the documentation changes have already been fixed in ab875cd65ff8d1013ec9014d2af9b7a737dc61d0 and e8af3768eda52dd0105a8a8bb0f460a2202a525a.
